radio/src/audio_arm.cpp

```diff
@@ -511,9 +515,16 @@ void audioTask(void * pdata)
 }
 #endif
 
-void mixSample(audio_data_t * result, int sample, unsigned int fade)
+void mixSample(audio_data_t * result, int sample, unsigned int fade, bool swVolumeAdjust = true)
 {
-  *result = limit(AUDIO_DATA_MIN, *result + ((sample >> fade) >> (16-AUDIO_BITS_PER_SAMPLE)), AUDIO_DATA_MAX);
+  *result = limit(AUDIO_DATA_MIN, *result + ((sample >> fade) >> (16 - AUDIO_BITS_PER_SAMPLE)), AUDIO_DATA_MAX);
+
+#if defined(SOFTWARE_VOLUME)
```

Comment:
The reason are the (Sine based) Tones. 
My initial attempt was to mix loop over the fully mixed buffer and apply volume on that but that caused tones to become cracks if volume is < 100%.
At least that's true with the Speaker delivered with the AR9X (2W 8Ohm).
As this is primarily for AR9X I considered it the "most important target for a good result" and adjusted it to work better.

I haven't debugged this to it's full extent but from looking at the code I do think that tone waves are kind of becoming to steep/irregular due to the interger-related fuzzyness when mixxing before volume control. 
The solution for that was to apply tone volume before mixing which in return meant this had to be done for for each wav sample, too.

But: It's well possible I simply did something wrong, I'd be glad to fix that up once I know what it is :)